tests/Spec.elm

```diff
@@ -69,6 +95,26 @@ suite =
           in Expect.equal
             (get (r.foo << try << r.bar) updatedExample)
             Nothing
+      , describe "dict"
+            [ test "set currently present to present" <| \_ ->
+                  let updatedDict = set (dictEntry "foo") (Just 9) dict
+                  in Expect.equal (get (dictEntry "foo") updatedDict) (Just 9)
+            , test "set currently absent to present" <| \_ ->
+                  let updatedDict = set (dictEntry "bar") (Just 9) dict
+                  in Expect.equal (get (dictEntry "bar") updatedDict) (Just 9)
+            , test "set currently present to absent" <| \_ ->
+                  let updatedDict = set (dictEntry "foo") Nothing dict
+                  in Expect.equal (get (dictEntry "foo") updatedDict) Nothing
+            , test "set currently absent to absent" <| \_ ->
+                  let updatedDict = set (dictEntry "bar") Nothing dict
+                  in Expect.equal (get (dictEntry "bar") updatedDict) Nothing
+            , test "set with try present" <| \_ ->
+                  let updatedDict = set (dictEntry "foo" << try << r.bar) "Sup" dictWithRecord
+                  in Expect.equal (get (dictEntry "foo" << try << r.bar) updatedDict) (Just "Sup")
+            , test "set with try absent" <| \_ ->
+                  let updatedDict = set (dictEntry "bar" << try << r.bar) "Sup" dictWithRecord
+                  in Expect.equal (get (dictEntry "bar" << try << r.bar) updatedDict) Nothing
+            ]
       ]
     , describe "over"
```

Comment:
I tried to minimize the combinatorial explosion of every composition for tests, but I can add more if desired.

Reply:
That's fine as it is :+1: